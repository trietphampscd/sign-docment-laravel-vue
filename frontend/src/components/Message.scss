
.message {
  position: fixed;
  right: 20px;
  bottom: 20px;
  .message-toggle-icon {
    position: fixed;
    right: 20px;
    bottom: 20px;
    z-index: 999;
    border-radius: 50%;
    width: 60px;
    height: 60px;
    background-color: #ef634c;
    display: flex;
    justify-content: center;
    align-items: center;

    &:hover{
      cursor: pointer;
      width: 64px;
      height: 64px;
      box-shadow: 0 0 7px 0 rgba(239, 99, 76, 0.5);
    }
  }

  .substraction__wrap {
    max-width: 350px;
    height: auto;
    background-color: #fcfafa;
    border: solid 1px #775649;
    border-radius: 5px;
    overflow: hidden;
    position: fixed;
    right: 20px;
    bottom: 100px;
    transform: scaleY(0);
    transition: -webkit-transform .3s;
    transition: transform .3s;
    transform-origin: bottom;
    z-index: 999;
    .substraction__title {
      padding: 20px 15px;
      background-image: url('/img/add_doc/group-71.png');
      background-repeat: no-repeat;
      background-position: top center;
      -webkit-background-size: cover;
      background-size: cover;
      color: #fff;
      .substraction__logo {
          margin-bottom: 15px;
          width: 85px;
          height: 30px;
      }
      .substraction__h3 {
          font-size: 22px;
          margin-bottom: 5px;
      } 
    }
    .substraction__form {
      max-width: 95%;
      margin: 10px auto;
      padding: 10px;
      -webkit-box-shadow: 0 0 6px #775649;
      box-shadow: 0 0 6px #775649;
      background-color: #fff;
      -webkit-border-radius: 4px;
      border-radius: 4px;

    }
  }
  
  .active{
    transform: scaleY(1);
    display: block;
  }

}
