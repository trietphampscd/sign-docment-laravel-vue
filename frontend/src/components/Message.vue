<template>
  <div class="message">
    <div class="message-toggle-icon" v-on:click="chat_room =!chat_room">
      <span v-if="!chat_room">
        <UserIcon icon="subtraction-icon.svg" />
      </span>
      <span v-if="chat_room">
        <UserIcon icon="close-icon.svg" />
      </span>
    </div>
    <div class="substraction__wrap" v-bind:class="chat_room?'active':''">
      <div class="substraction__title">
        <div class="substraction__logo">
          <img src="img/logo_white.svg" alt="Coffee Sign" width="120" />
        </div>
        <h3 class="substraction__h3">Hi there! ðŸ‘‹</h3>
        <p
          class="substraction__text"
        >Please fill out the form below to start chatting with the next available agent.</p>
      </div>
      <!-- /.substraction__title -->
      <form action="#" class="substraction__form">
        <b-form-group>
          <b-input-group>
            <b-input-group-prepend>
              <b-input-group-text>
                <i class="fa fa-user"></i>
              </b-input-group-text>
            </b-input-group-prepend>
            <b-form-input type="text" placeholder="Recipient Name*"></b-form-input>
          </b-input-group>
        </b-form-group>
        <b-form-group>
          <b-input-group>
            <b-input-group-prepend>
              <b-input-group-text>
                <i class="fa fa-envelope"></i>
              </b-input-group-text>
            </b-input-group-prepend>
            <b-form-input type="email" placeholder="Email Address*" autocomplete="email"></b-form-input>
          </b-input-group>
        </b-form-group>
        <b-form-group>
          <b-input-group>
            <b-input-group-prepend>
              <b-input-group-text>
                <i class="fa fa-commenting"></i>
              </b-input-group-text>
            </b-input-group-prepend>
            <b-form-textarea type="text" placeholder="Enter Message" row="5" class="email-msg"></b-form-textarea>
          </b-input-group>
        </b-form-group>
        <b-button type="submit" block variant="primary">Start Chat</b-button>
      </form>
      <!-- /.substraction__form -->

      <b-button variant="link" block>
        We're by
        <b>tawk.to</b>
      </b-button>
    </div>
  </div>
</template>

<script>
import UserIcon from "./UserIcon";
export default {
  name: "Message",
  components: {
    UserIcon
  },
  data() {
    return {
      chat_room: false
    };
  },
  methods: {}
};
</script>
<style lang="scss">
@import "./Message.scss";
</style>
