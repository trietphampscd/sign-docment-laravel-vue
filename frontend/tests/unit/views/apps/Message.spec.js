import Vue from 'vue'
import { shallowMount, mount } from '@vue/test-utils'
import BootstrapVue from 'bootstrap-vue'
import Message from '@/views/apps/email/Message'

Vue.use(BootstrapVue)

describe('Message.vue', () => {
  it('has a name', () => {
    expect(Message.name).toMatch('message')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Message)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Message', () => {
    const wrapper = shallowMount(Message)
    expect(wrapper.is(Message)).toBe(true)
  })
  it('should render correct content', () => {
    const wrapper = mount(Message)
    expect(wrapper.find('div.email-app > nav > a.btn').text()).toMatch('New Email')
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Message)
    expect(wrapper.element).toMatchSnapshot()
  })
})
