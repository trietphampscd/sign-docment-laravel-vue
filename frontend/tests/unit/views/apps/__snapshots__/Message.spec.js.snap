// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message.vue renders correctly 1`] = `
<div
  class="animated fadeIn"
>
  <div
    class="email-app mb-4"
  >
    <nav>
      <a
        class="btn btn-danger btn-block"
        href="#/apps/email/compose"
      >
        New Email
      </a>
       
      <ul
        class="nav"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="#"
          >
            <i
              class="fa fa-inbox"
            />
             Inbox 
            <bbadge-stub
              activeclass="active"
              event="click"
              exactactiveclass="active"
              routertag="a"
              tag="span"
              target="_self"
              variant="danger"
            >
              4
            </bbadge-stub>
          </a>
        </li>
         
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="#"
          >
            <i
              class="fa fa-star"
            />
             Stared
          </a>
        </li>
         
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="#"
          >
            <i
              class="fa fa-rocket"
            />
             Sent
          </a>
        </li>
         
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="#"
          >
            <i
              class="fa fa-trash-o"
            />
             Trash
          </a>
        </li>
         
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="#"
          >
            <i
              class="fa fa-bookmark"
            />
             Important
            <bbadge-stub
              activeclass="active"
              event="click"
              exactactiveclass="active"
              routertag="a"
              tag="span"
              target="_self"
              variant="info"
            >
              5
            </bbadge-stub>
          </a>
        </li>
         
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="#"
          >
            <i
              class="fa fa-inbox"
            />
             Inbox 
            <bbadge-stub
              activeclass="active"
              event="click"
              exactactiveclass="active"
              routertag="a"
              tag="span"
              target="_self"
              variant="danger"
            >
              4
            </bbadge-stub>
          </a>
        </li>
      </ul>
    </nav>
     
    <main
      class="message"
    >
      <div
        class="toolbar"
      >
        <div
          class="btn-group mr-1"
        >
          <bbutton-stub
            activeclass="active"
            event="click"
            exactactiveclass="active"
            routertag="a"
            target="_self"
            type="button"
            variant="light"
          >
            <span
              class="fa fa-star"
            />
          </bbutton-stub>
           
          <bbutton-stub
            activeclass="active"
            event="click"
            exactactiveclass="active"
            routertag="a"
            target="_self"
            type="button"
            variant="light"
          >
            <span
              class="fa fa-star-o"
            />
          </bbutton-stub>
           
          <bbutton-stub
            activeclass="active"
            event="click"
            exactactiveclass="active"
            routertag="a"
            target="_self"
            type="button"
            variant="light"
          >
            <span
              class="fa fa-bookmark-o"
            />
          </bbutton-stub>
        </div>
         
        <div
          class="btn-group mr-1"
        >
          <bbutton-stub
            activeclass="active"
            event="click"
            exactactiveclass="active"
            routertag="a"
            target="_self"
            type="button"
            variant="light"
          >
            <span
              class="fa fa-mail-reply"
            />
          </bbutton-stub>
           
          <bbutton-stub
            activeclass="active"
            event="click"
            exactactiveclass="active"
            routertag="a"
            target="_self"
            type="button"
            variant="light"
          >
            <span
              class="fa fa-mail-reply-all"
            />
          </bbutton-stub>
           
          <bbutton-stub
            activeclass="active"
            event="click"
            exactactiveclass="active"
            routertag="a"
            target="_self"
            type="button"
            variant="light"
          >
            <span
              class="fa fa-mail-forward"
            />
          </bbutton-stub>
        </div>
         
        <bbutton-stub
          activeclass="active"
          event="click"
          exactactiveclass="active"
          routertag="a"
          target="_self"
          type="button"
          variant="light"
        >
          <span
            class="fa fa-trash-o"
          />
        </bbutton-stub>
         
        <bdropdown-stub
          boundary="scrollParent"
          class="ml-1"
          role="menu"
          toggletext="Toggle Dropdown"
          variant="light"
        >
          <template>
            <span
              class="fa fa-tags"
            />
          </template>
           
          <bdropdownitem-stub
            activeclass="active"
            event="click"
            exactactiveclass="active"
            href="#"
            routertag="a"
            target="_self"
          >
            add label 
            <bbadge-stub
              activeclass="active"
              event="click"
              exactactiveclass="active"
              routertag="a"
              tag="span"
              target="_self"
              variant="danger"
            >
               Home
            </bbadge-stub>
          </bdropdownitem-stub>
           
          <bdropdownitem-stub
            activeclass="active"
            event="click"
            exactactiveclass="active"
            href="#"
            routertag="a"
            target="_self"
          >
            add label 
            <bbadge-stub
              activeclass="active"
              event="click"
              exactactiveclass="active"
              routertag="a"
              tag="span"
              target="_self"
              variant="info"
            >
               Job
            </bbadge-stub>
          </bdropdownitem-stub>
           
          <bdropdownitem-stub
            activeclass="active"
            event="click"
            exactactiveclass="active"
            href="#"
            routertag="a"
            target="_self"
          >
            add label 
            <bbadge-stub
              activeclass="active"
              event="click"
              exactactiveclass="active"
              routertag="a"
              tag="span"
              target="_self"
              variant="success"
            >
               Clients
            </bbadge-stub>
          </bdropdownitem-stub>
           
          <bdropdownitem-stub
            activeclass="active"
            event="click"
            exactactiveclass="active"
            href="#"
            routertag="a"
            target="_self"
          >
            add label 
            <bbadge-stub
              activeclass="active"
              event="click"
              exactactiveclass="active"
              routertag="a"
              tag="span"
              target="_self"
              variant="warning"
            >
               News
            </bbadge-stub>
          </bdropdownitem-stub>
        </bdropdown-stub>
      </div>
       
      <div
        class="details"
      >
        <div
          class="title"
        >
          Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit.
        </div>
         
        <div
          class="header"
        >
          <img
            class="avatar"
            src="img/avatars/7.jpg"
          />
           
          <div
            class="from"
          >
            <span>
              Lukasz Holeczek
            </span>
            
            lukasz@bootstrapmaster.com
          
          </div>
           
          <div
            class="date"
          >
            Today, 
            <b>
              3:47 PM
            </b>
          </div>
        </div>
         
        <div
          class="content"
        >
          <p>
            
            Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          
          </p>
           
          <blockquote>
            
            Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          
          </blockquote>
        </div>
         
        <div
          class="attachments"
        >
          <div
            class="attachment"
          >
            <bbadge-stub
              activeclass="active"
              event="click"
              exactactiveclass="active"
              routertag="a"
              tag="span"
              target="_self"
              variant="danger"
            >
              zip
            </bbadge-stub>
             
            <b>
              bootstrap.zip
            </b>
             
            <i>
              (2,5MB)
            </i>
             
            <span
              class="menu"
            >
              <a
                class="fa fa-search"
                href="#"
              />
               
              <a
                class="fa fa-share"
                href="#"
              />
               
              <a
                class="fa fa-cloud-download"
                href="#"
              />
            </span>
          </div>
           
          <div
            class="attachment"
          >
            <bbadge-stub
              activeclass="active"
              event="click"
              exactactiveclass="active"
              routertag="a"
              tag="span"
              target="_self"
              variant="info"
            >
              txt
            </bbadge-stub>
             
            <b>
              readme.txt
            </b>
             
            <i>
              (7KB)
            </i>
             
            <span
              class="menu"
            >
              <a
                class="fa fa-search"
                href="#"
              />
               
              <a
                class="fa fa-share"
                href="#"
              />
               
              <a
                class="fa fa-cloud-download"
                href="#"
              />
            </span>
          </div>
           
          <div
            class="attachment"
          >
            <bbadge-stub
              activeclass="active"
              event="click"
              exactactiveclass="active"
              routertag="a"
              tag="span"
              target="_self"
              variant="success"
            >
              xls
            </bbadge-stub>
             
            <b>
              spreadsheet.xls
            </b>
             
            <i>
              (984KB)
            </i>
             
            <span
              class="menu"
            >
              <a
                class="fa fa-search"
                href="#"
              />
               
              <a
                class="fa fa-share"
                href="#"
              />
               
              <a
                class="fa fa-cloud-download"
                href="#"
              />
            </span>
          </div>
        </div>
         
        <form
          action=""
          method="post"
        >
          <div
            class="form-group"
          >
            <textarea
              class="form-control"
              id="message"
              name="body"
              placeholder="Click here to reply"
              rows="12"
            />
          </div>
           
          <div
            class="form-group"
          >
            <bbutton-stub
              activeclass="active"
              event="click"
              exactactiveclass="active"
              routertag="a"
              tabindex="3"
              target="_self"
              type="submit"
              variant="success"
            >
              Send message
            </bbutton-stub>
          </div>
        </form>
      </div>
    </main>
  </div>
</div>
`;
